### PR TITLE
PARQUET-503: Reenable parquet 2.0 encoding implementations.

### PR DESCRIPTION
**Pull request by nongli**

### CODE REVIEW

File: src/parquet/util/bit-stream-utils.inline.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ inline bool BitWriter::PutAligned(T val, int num_bytes) {
   return true;
 }
 
-inline bool BitWriter::PutVlqInt(int32_t v) {
+inline bool BitWriter::PutVlqInt(uint32_t v) {
```
Comment (wesm) [inline src/parquet/util/bit-stream-utils.inline.h]: does this have any correctness implications? I assume that this function only ever was getting used with non-negative integers

**Comment by wesm** [comment]
lgtm

File: src/parquet/util/bit-stream-utils.inline.h (hunk 1 above)
Comment (nongli) [inline src/parquet/util/bit-stream-utils.inline.h]: I think the impala implementation is wrong and doesn't handle negative numbers. We want to do a logical shift, not arithmetic shift. The version with int32_t will spin forever for negative numbers.

Comment (julienledem) [inline src/parquet/util/bit-stream-utils.inline.h]: FYI: the java side deals with signed integers

**Comment by julienledem** [comment]
+1

File: src/parquet/util/bit-stream-utils.inline.h (hunk 1 above)
Comment (nongli) [inline src/parquet/util/bit-stream-utils.inline.h]: I think this just does the right thing. The implicit cast from signed to unsigned is correct.

**Pull request closed without merging**